addons/event_sale/models/sale_order.py

```diff
@@ -18,20 +18,14 @@ def write(self, vals):
             registrations_toupdate.write({'partner_id': vals['partner_id']})
         return result
 
-    def _action_confirm(self):
-        res = super(SaleOrder, self)._action_confirm()
-        for so in self:
-            # confirm registration if it was free (otherwise it will be confirmed once invoice fully paid)
-            so.order_line._update_registrations(confirm=so.amount_total == 0, cancel_to_draft=False)
-        return res
-
     def action_confirm(self):
+        if not self.env.context.get('registrations_confirmed'):
+            for so in self:
+                if any(so.order_line.filtered(lambda line: line.event_id)):
+                    return self.env['ir.actions.act_window'] \
+                        .with_context(default_sale_order_id=so.id) \
+                        .for_xml_id('event_sale', 'action_sale_order_event_registration')
         res = super(SaleOrder, self).action_confirm()
-        for so in self:
-            if any(so.order_line.filtered(lambda line: line.event_id)):
-                return self.env['ir.actions.act_window'] \
-                    .with_context(default_sale_order_id=so.id) \
-                    .for_xml_id('event_sale', 'action_sale_order_event_registration')
```

Comment:
Well not sure you don't break things ... I understand the need. However by returning the modal before calling super, you break all flows that use action_confirm and do not know they have to use registrations_confirmed set to True to bypass the code.

Beforehand, people could call action_confirm and ignore the returned result. It means their SO was confirmed with all implied things, and not using the modal was not a big deal. Now if they don't update their code very carefully it may break.

I think you should not change that logic, which is already a bit strange.

As you have anyway a hook on write of registrations to update lead values, is it really an issue ?